inference-engine/src/cldnn_engine/cldnn_engine.cpp

```diff
@@ -80,146 +83,114 @@ cldnn::device_info clDNNEngine::GetDeviceInfo(const std::map<std::string, std::s
     return device_info;
 }
 
-InferenceEngine::ICNNNetwork::Ptr clDNNEngine::CloneAndTransformNetwork(const InferenceEngine::ICNNNetwork& network, CLDNNPlugin::Config config) const {
-    std::shared_ptr<ICNNNetwork> clonedNetwork = cloneNetwork(network);
-    bool baselineIsFP16 = false;
+template<typename T>
+static bool disableReduceDecomposition(const std::shared_ptr<const ngraph::Node> node) {
+    if (auto op = std::dynamic_pointer_cast<const T>(node)) {
+        auto reduction_axes = op->get_reduction_axes().to_vector();
+        bool reduce_along_f = op->get_reduction_axes().size() == 1 && std::count(reduction_axes.begin(), reduction_axes.end(), 1) != 0;
+        bool fp16_batch_not_1 = op->get_element_type() == ngraph::element::f16 && op->input(0).get_shape()[0] != 1;
+        bool can_use_reduce = !reduce_along_f && !fp16_batch_not_1;
+        return can_use_reduce;
+    }
+    return false;
+}
 
+InferenceEngine::ICNNNetwork::Ptr clDNNEngine::CloneAndTransformNetwork(const InferenceEngine::ICNNNetwork& network, const CLDNNPlugin::Config& config) const {
+    std::shared_ptr<ICNNNetwork> clonedNetwork = std::make_shared<CNNNetworkNGraphImpl>(network);
```

Comment:
I think we need to add simple function to clone ngraph impl, since current `CloneNetwork` is in legacy library and can handle old representation as well. 
New small function should be a part of Plugin API. CC @apankratovantonp such function is supposed to be used in HETERO / MULTI if we need network copy here.

CC @ilyachur

Reply:
@ilyachur do you think the `ie_ngraph_utils.hpp` a proper file to adding this small `cloneFunction` function? I think we can add:
```c++
std::shared_ptr<ICNNNetwork> InferenceEngine::details::cloneNetwork(const ICNNNetwork& network) {
    OV_ITT_SCOPED_TASK(itt::domains::IE, "cloneNetwork");

    if (network.getFunction()) {
        return std::make_shared<details::CNNNetworkNGraphImpl>(network);
    }

    THROW_IE_EXCEPTION << "InferenceEngine::details::cloneNetwork requires ngraph-based `network` object to clone";
}
```
With implementation in `cnn_network_ngraph_impl.cpp`

Reply:
@ilyachur  ping

Reply:
@ilya-lavrenov Yes I think we can do it.

**Just to highlight:**
We need to have cloneNetwork only because we have some additional information about pre-processing and input precisions in the CNNNetwork.
If nGraph function contain all this information we can use `clone_function`

Reply:
Added in #3450